File: server/java/src/main/java/com/onlinepayments/example/service/domain/paymentdetails/PaymentStatusOutput.java
package com.onlinepayments.example.service.domain.paymentdetails;

import java.util.List;

public record PaymentStatusOutput(
    List<APIError> errors,
    Boolean isAuthorized,
    Boolean isCancellable,
    Boolean isRefundable,
    String statusCategory,
    Integer statusCode,
    String statusCodeChangeDateTime
) { }

File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentStatusOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentStatusOutput
{
    /** @var APIError[]|null */
    public ?array $errors = null;
    public ?bool $isAuthorized = null;
    public ?bool $isCancellable = null;
    public ?bool $isRefundable = null;
    public ?string $statusCategory = null;
    public ?int $statusCode = null;
    public ?string $statusCodeChangeDateTime = null;

    public function toArray(): array
    {
        return [
            'errors' => $this->errors ? array_map(fn($e) => $e->toArray(), $this->errors) : null,
            'isAuthorized' => $this->isAuthorized,
            'isCancellable' => $this->isCancellable,
            'isRefundable' => $this->isRefundable,
            'statusCategory' => $this->statusCategory,
            'statusCode' => $this->statusCode,
            'statusCodeChangeDateTime' => $this->statusCodeChangeDateTime,
        ];
    }
}
